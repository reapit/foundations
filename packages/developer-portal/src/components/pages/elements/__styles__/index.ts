import { css } from 'linaria'

export const iframeWrapper = css`
  height: 88.5vh;
  width: 100%;
`
