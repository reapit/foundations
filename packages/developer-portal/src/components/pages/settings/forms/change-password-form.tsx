import React from 'react'
import {
  FormSection,
  FormSubHeading,
  Input,
  Button,
  Form,
  Formik,
  FormHeading,
  LevelRight,
  Grid,
  GridItem,
  FormikHelpers,
} from '@reapit/elements'
import { validationSchemaChangePassword as validationSchema } from './form-schema/validation-schema'
import { formFieldsChangePassword } from './form-schema/form-fields'

const { currentPasswordField, passwordField, confirmPasswordField } = formFieldsChangePassword

export type ChangePasswordValues = {
  currentPassword: string
  password: string
  confirmPassword: string
}

export type ChangePasswordFormProps = {
  changePassword: (values: ChangePasswordValues) => void
}

export const handleSubmitChangePassword = (changePassword: (values: ChangePasswordValues) => void) => (
  values: ChangePasswordValues,
  { setSubmitting }: FormikHelpers<ChangePasswordValues>,
) => {
  setSubmitting(true)
  changePassword(values)
}

export const ChangePasswordForm: React.FC<ChangePasswordFormProps> = ({ changePassword }) => {
  return (
    <FormSection>
      <Formik
        initialValues={{
          currentPassword: '',
          password: '',
          confirmPassword: '',
        }}
        validationSchema={validationSchema}
        onSubmit={handleSubmitChangePassword(changePassword)}
      >
        {({ isSubmitting, isValidating, isValid, touched }) => {
          const isEnable =
            isValid && Boolean(touched) && (touched.confirmPassword || touched.currentPassword || touched.password)

          return (
            <Form>
              <FormHeading>Change password</FormHeading>
              <FormSubHeading>
                Please complete the following fields to change your password. You will be automatically logged out for
                the changes to be applied
              </FormSubHeading>
              <Grid>
                <GridItem>
                  <Input
                    dataTest="current-password"
                    type="password"
                    labelText={currentPasswordField.label}
                    id={currentPasswordField.name}
                    name={currentPasswordField.name}
                  />
                </GridItem>
                <GridItem>
                  <Input
                    dataTest="password"
                    type="password"
                    labelText={passwordField.label}
                    id={passwordField.name}
                    name={passwordField.name}
                  />
                </GridItem>
              </Grid>
              <Grid>
                <GridItem className="is-half">
                  <Input
                    dataTest="confirmPassword"
                    type="password"
                    labelText={confirmPasswordField.label}
                    id={confirmPasswordField.name}
                    name={confirmPasswordField.name}
                  />
                </GridItem>
              </Grid>
              <LevelRight>
                <Button
                  dataTest="button-change-password"
                  disabled={!isEnable}
                  loading={isSubmitting || isValidating}
                  variant="primary"
                  type="submit"
                >
                  Change Password
                </Button>
              </LevelRight>
            </Form>
          )
        }}
      </Formik>
    </FormSection>
  )
}

export default ChangePasswordForm
