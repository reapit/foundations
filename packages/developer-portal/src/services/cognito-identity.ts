import { CognitoUserPool, CognitoUser, AuthenticationDetails } from 'amazon-cognito-identity-js'
import { logger } from '@reapit/utils'

export const getNewUser = (userName: string, connectClientId: string, userPoolId?: string) => {
  const poolData = {
    UserPoolId: userPoolId || window?.reapit?.config?.connectUserPoolId,
    ClientId: connectClientId,
  }
  const userPool = new CognitoUserPool(poolData)
  const userData = {
    Username: userName,
    Pool: userPool,
  }
  return new CognitoUser(userData)
}

export interface ChangePasswordParams {
  newPassword: string
  userName: string
  password: string
  connectClientId: string
}

export interface ConfirmRegistrationParams {
  userName: string
  verificationCode: string
  connectClientId: string
}

export const changePasswordService = async ({
  password,
  userName,
  newPassword,
  connectClientId,
}: ChangePasswordParams): Promise<string> => {
  return new Promise((resolve, reject) => {
    const authenticationData = {
      Username: userName,
      Password: password,
    }
    const authenticationDetails = new AuthenticationDetails(authenticationData)
    const cognitoUser = getNewUser(userName, connectClientId)
    cognitoUser.authenticateUser(authenticationDetails, {
      onSuccess: () => {
        cognitoUser.changePassword(password, newPassword, (err, result) => {
          if (err) {
            logger(new Error(err.message))
            reject(err)
          }
          resolve(result as string)
        })
      },
      onFailure: err => {
        logger(new Error(err.message))
        reject(err)
      },
    })
  })
}

export const confirmRegistrationService = async ({
  verificationCode,
  userName,
  connectClientId,
}: ConfirmRegistrationParams): Promise<string> => {
  return new Promise((resolve, reject) => {
    const cognitoUser = getNewUser(userName, connectClientId)

    cognitoUser.confirmRegistration(verificationCode, true, err => {
      if (err) {
        logger(new Error(err.message))
        reject(`Bad request, confirm registration service failed ${JSON.stringify(err)}`)
      }
      resolve('SUCCESS')
    })
  })
}

export const confirmRegistration = async (params: ConfirmRegistrationParams): Promise<string | undefined> => {
  const { verificationCode, userName, connectClientId } = params
  const paramsValid = verificationCode && userName && connectClientId

  try {
    if (!paramsValid) {
      throw new Error('Bad request, verification code and username are required')
    }
    return await confirmRegistrationService(params)
  } catch (err) {
    logger(new Error(err.message))
    console.error(`Bad request, failed to confirm registration, ${err}`)
  }
}
