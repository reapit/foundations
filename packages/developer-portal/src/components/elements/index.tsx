import {
  SmallText,
  Button,
  elHFull,
  elMb5,
  FlexContainer,
  Icon,
  PageContainer,
  SecondaryNavContainer,
  BodyText,
  Title,
} from '@reapit/elements'
import React, { FC } from 'react'
import { openNewPage, ExternalPages } from '../../utils/navigation'
import { iframeWrapper } from './__styles__/index'

export const ElementsPage: FC = () => {
  return (
    <FlexContainer isFlexAuto>
      <SecondaryNavContainer>
        <Icon className={elMb5} icon="elementsInfographic" iconSize="large" />
        <BodyText>Reapit Elements</BodyText>
        <SmallText hasGreyText>
          Elements is a UI toolkit built in React JS, exporting a library of CSS classes to help you build clean,
          interactive user interfaces for the Reapit Foundations ecosystem. It is based on the Reapit Foundations Design
          System, our in house styleguide.
        </SmallText>
        <Button className={elMb5} intent="default" onClick={openNewPage(ExternalPages.elementsDocs)}>
          View Docs
        </Button>
        <Button className={elMb5} intent="primary" onClick={openNewPage(process.env.elementsUri)}>
          Storybook
        </Button>
      </SecondaryNavContainer>
      <PageContainer className={elHFull}>
        <Title>Elements</Title>
        <iframe
          className={iframeWrapper}
          src={process.env.elementsUri}
          frameBorder="0"
          allow="clipboard-write"
        ></iframe>
      </PageContainer>
    </FlexContainer>
  )
}

export default ElementsPage
