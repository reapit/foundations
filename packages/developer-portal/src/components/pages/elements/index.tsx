import {
  BodyText,
  Button,
  elHFull,
  elMb3,
  elMb8,
  FlexContainer,
  Icon,
  PageContainer,
  SecondaryNav,
  SecondaryNavContainer,
  SecondaryNavItem,
  Subtitle,
  Title,
} from '@reapit/elements'
import React, { FC } from 'react'
import { useHistory, useLocation } from 'react-router'
import Routes from '../../../constants/routes'
import { openNewPage, ExternalPages, navigate } from '../../../utils/navigation'
import { iframeWrapper } from './__styles__/index'

export const ElementsPage: FC = () => {
  const location = useLocation()
  const history = useHistory()
  const { pathname } = location
  return (
    <FlexContainer isFlexAuto>
      <SecondaryNavContainer>
        <Title>Docs</Title>
        <SecondaryNav className={elMb8}>
          <SecondaryNavItem onClick={navigate(history, Routes.API_DOCS)} active={pathname === Routes.API_DOCS}>
            API Docs
          </SecondaryNavItem>
          <SecondaryNavItem onClick={navigate(history, Routes.ELEMENTS)} active={pathname === Routes.ELEMENTS}>
            Elements
          </SecondaryNavItem>
        </SecondaryNav>
        <Icon icon="elementsInfographic" iconSize="large" />
        <Subtitle>Reapit Elements</Subtitle>
        <BodyText hasGreyText>
          Elements is a UI toolkit built in React JS, exporting a library of CSS classes to help you build clean,
          interactive user interfaces for the Reapit Foundations ecosystem. It is based on the Reapit Foundations Design
          System, our in house styleguide.
        </BodyText>
        <Button className={elMb3} intent="neutral" onClick={openNewPage(ExternalPages.elementsDocs)}>
          View Docs
        </Button>
        <Button className={elMb3} intent="critical" onClick={openNewPage(window.reapit.config.elementsUri)}>
          Open New
        </Button>
      </SecondaryNavContainer>
      <PageContainer className={elHFull}>
        <Title>Elements</Title>
        <iframe
          className={iframeWrapper}
          src={window.reapit.config.elementsUri}
          frameBorder="0"
          allow="clipboard-write"
        ></iframe>
      </PageContainer>
    </FlexContainer>
  )
}

export default ElementsPage
