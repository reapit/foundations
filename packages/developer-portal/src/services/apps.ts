import {
  CreateAppModel,
  PagedResultAppSummaryModel_,
  AppDetailModel,
  PagedResultAppRevisionModel_,
  CreateAppRevisionModel,
  AppRevisionModel,
  ApproveModel,
  RejectRevisionModel,
  AppClientSecretModel,
} from '@reapit/foundations-ts-definitions'
import { fetcher, fetcherWithReturnHeader, fetcherWithRawUrl, setQueryParams } from '@reapit/elements'
import { URLS } from './constants'
import { generateHeader } from './utils'
import { logger } from '@reapit/utils'
import { FetchByIdCommonParams, FetchListCommonParams } from './types'

export type FetchAppsListParams = FetchListCommonParams & {
  developerId?: string[]
  clientId?: string
  externalAppId?: string[]
  category?: string[]
  desktopIntegrationTypeId?: string
  appName?: string
  developerName?: string
  companyName?: string
  isFeatured?: boolean
  isDirectApi?: boolean
  onlyInstalled?: boolean
  registeredFrom?: string
  registeredTo?: string
}

export type FetchAppByIdParams = FetchByIdCommonParams & {
  clientId?: string
}

export type CreateAppParams = CreateAppModel & {
  successCallback?: (appDetail: AppDetailModel) => void
}

export type DeleteAppByIdParams = FetchByIdCommonParams & {
  successCallback?: () => void
}

export type FeatureAppByIdParams = FetchByIdCommonParams

export type UnfeatureAppByIdParams = FetchByIdCommonParams

export type FetchAppRevisionsListParams = FetchListCommonParams & {
  id: string
}

export type CreateAppRevisionParams = FetchByIdCommonParams & CreateAppRevisionModel

export type FetchAppRevisionsByIdParams = FetchByIdCommonParams & {
  revisionId: string
}

export type ApproveAppRevisionByIdParams = FetchByIdCommonParams & { revisionId: string } & ApproveModel

export type RejectAppRevisionByIdParams = FetchByIdCommonParams & { revisionId: string } & RejectRevisionModel

export type FetchAppSecretByIdParams = FetchByIdCommonParams

export const fetchAppsListAPI = async (params: FetchAppsListParams): Promise<PagedResultAppSummaryModel_> => {
  try {
    const response = await fetcher({
      url: `${URLS.apps}?${setQueryParams(params)}`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'GET',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw error?.response
  }
}

export const fetchAppByIdByRawUrl = async (url: string): Promise<AppDetailModel> => {
  try {
    const response = await fetcherWithRawUrl({
      url,
      method: 'GET',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })

    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const fetchAppById = async (params: FetchAppByIdParams): Promise<AppDetailModel> => {
  try {
    const { id, clientId } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}?${setQueryParams({ clientId })}`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'GET',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const createAppAPI = async (params: CreateAppParams) => {
  try {
    const headers = await fetcherWithReturnHeader({
      url: URLS.apps,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'POST',
      body: params,
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return headers
  } catch (error) {
    logger(error)
    throw error?.response
  }
}

export const deleteAppById = async (params: DeleteAppByIdParams) => {
  try {
    const { id } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'DELETE',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw error?.response
  }
}

export const featureAppById = async (params: FeatureAppByIdParams) => {
  try {
    const { id } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/feature`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'PUT',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const unfeatureAppById = async (params: UnfeatureAppByIdParams) => {
  try {
    const { id } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/feature`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'DELETE',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const fetchAppRevisionsList = async (
  params: FetchAppRevisionsListParams,
): Promise<PagedResultAppRevisionModel_> => {
  try {
    const { id, ...rest } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/revisions?${setQueryParams(rest)}`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'GET',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const createAppRevision = async (params: CreateAppRevisionParams) => {
  try {
    const { id, ...rest } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/revisions`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'POST',
      body: rest,
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw error
  }
}

export const fetchAppRevisionsById = async (params: FetchAppRevisionsByIdParams): Promise<AppRevisionModel> => {
  try {
    const { id, revisionId } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/revisions/${revisionId}`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'GET',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw error?.response
  }
}

export const approveAppRevisionById = async (params: ApproveAppRevisionByIdParams) => {
  try {
    const { id, revisionId, ...rest } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/revisions/${revisionId}/approve`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'POST',
      body: rest,
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const rejectAppRevisionById = async (params: RejectAppRevisionByIdParams) => {
  try {
    const { id, revisionId, ...rest } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/revisions/${revisionId}/reject`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'POST',
      body: rest,
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const fetchAppSecretByIdAPI = async (params: FetchAppSecretByIdParams): Promise<AppClientSecretModel> => {
  try {
    const { id } = params
    const response = await fetcher({
      url: `${URLS.apps}/${id}/secret`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'GET',
      headers: generateHeader(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw error?.response
  }
}

export const fetchDesktopIntegrationTypes = async () => {
  const response = await fetcher({
    url: URLS.desktopIntegrationTypes,
    method: 'GET',
    api: window.reapit.config.marketplaceApiUrl,
    headers: generateHeader(window.reapit.config.marketplaceApiKey),
  })
  return response
}

export const fetchAppDetail = async ({ clientId, id }) => {
  const response = await fetcher({
    url: clientId ? `${URLS.apps}/${id}?clientId=${clientId}` : `${URLS.apps}/${id}`,
    api: window.reapit.config.marketplaceApiUrl,
    method: 'GET',
    headers: generateHeader(window.reapit.config.marketplaceApiKey),
  })
  return response
}

export const fetchAppApiKey = async ({ installationId }) => {
  const response = await fetcher({
    url: `${URLS.installations}/${installationId}/apiKey`,
    api: window.reapit.config.marketplaceApiUrl,
    method: 'GET',
    headers: generateHeader(window.reapit.config.marketplaceApiKey),
  })
  return response
}
