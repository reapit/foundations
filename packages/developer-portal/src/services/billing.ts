// import { BillingS } from '@reapit/foundations-ts-definitions'
import { fetcher, setQueryParams } from '@reapit/elements'
import { generateHeaderWithApiV2 } from './utils'
import { URLS } from './constants'
import { logger } from '@reapit/utils'
import { BillingBreakdownForMonthV2Model } from '@reapit/foundations-ts-definitions'

// Manual defined Model
export type BillingSummaryModel = {
  description?: string
  from?: string
  to?: string
  requestsByPeriod?: MonthlyBillingDetailsModel
}

export type MonthlyBillingDetailsModel = {
  period?: string
  periodStart?: string
  periodEnd?: string
  periodName?: string
  requestCount?: number
  endpointCount?: number
  netAmount?: number
  grossAmount?: number
  vatAmount?: number
}

export type BillingDetailsModel = {
  period?: string
  totalEndpoints?: number
  totalRequests?: number
  totalCost?: number
  requestsByService?: ServiceBillingDetailsModel[]
}

export type ServiceBillingDetailsModel = {
  serviceName?: string
  endpointCount?: number
  requestCount?: number
  cost?: number
  requestsByEndpoint?: EndpointBillingDetailsModel[]
}

export type EndpointBillingDetailsModel = {
  endpoint?: string
  requestCount?: number
  cost?: number
}

export type FetchBillingsParams = {
  developerId?: string
  dateFrom?: string
  dateTo?: string
}

export type FetchBillingsByMonthParams = {
  developerId?: string
  month: string
}

export const fetchBillings = async (params: FetchBillingsParams): Promise<BillingSummaryModel> => {
  try {
    const response = await fetcher({
      url: `${URLS.trafficEventBilling}?${setQueryParams(params)}`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'GET',
      headers: generateHeaderWithApiV2(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}

export const fetchBillingsByMonth = async (
  params: FetchBillingsByMonthParams,
): Promise<BillingBreakdownForMonthV2Model> => {
  try {
    const { month, ...rest } = params
    const response = await fetcher({
      url: `${URLS.trafficEventBilling}/${month}?${setQueryParams(rest)}`,
      api: window.reapit.config.marketplaceApiUrl,
      method: 'GET',
      headers: generateHeaderWithApiV2(window.reapit.config.marketplaceApiKey),
    })
    return response
  } catch (error) {
    logger(error)
    throw new Error(error)
  }
}
