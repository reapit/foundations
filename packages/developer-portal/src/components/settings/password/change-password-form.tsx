import React, { FC } from 'react'
import {
  BodyText,
  Button,
  ButtonGroup,
  elMb11,
  FormLayout,
  InputGroup,
  InputWrap,
  Loader,
  UseSnack,
  useSnack,
} from '@reapit/elements'
import { useReapitConnect } from '@reapit/connect-session'
import { reapitConnectBrowserSession } from '../../../core/connect-session'
import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import { validationSchemaChangePassword } from './validation-schema'
import { changePasswordService } from '../../../services/cognito-identity'
import { openNewPage } from '../../../utils/navigation'
import { getTokenIssuer, tokenFromCognito } from '../../../utils/token'

export type ChangePasswordFormValues = {
  password: string
  newPassword: string
  confirmPassword: string
}

export const handleChangePassword =
  (email: string, { success, error }: UseSnack) =>
  async ({ newPassword, password }: ChangePasswordFormValues) => {
    const passwordChanged = await changePasswordService({
      password,
      newPassword,
      userName: email,
    })

    if (passwordChanged) {
      success('Successfully updated your password')
    } else {
      error('Failed to update your password. This error has been logged, please try again')
    }
  }

export const ChangePasswordForm: FC = () => {
  const { connectSession } = useReapitConnect(reapitConnectBrowserSession)
  const snacks = useSnack()
  const email = connectSession?.loginIdentity.email ?? ''
  const token = connectSession?.accessToken ?? ''

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<ChangePasswordFormValues>({
    resolver: yupResolver(validationSchemaChangePassword),
    defaultValues: {
      password: '',
      newPassword: '',
      confirmPassword: '',
    },
  })

  if (!token) return <Loader />

  if (tokenFromCognito(token)) {
    return (
      <form onSubmit={handleSubmit(handleChangePassword(email, snacks))}>
        <FormLayout hasMargin>
          <InputWrap>
            <InputGroup
              {...register('password')}
              type="password"
              label="Current Password"
              placeholder="Current Password"
              errorMessage={errors?.password?.message}
              icon={errors?.password?.message ? 'asterisk' : null}
              intent="danger"
            />
          </InputWrap>
          <InputWrap>
            <InputGroup
              {...register('newPassword')}
              type="password"
              label="New Password"
              placeholder="New Password"
              errorMessage={errors?.newPassword?.message}
              icon={errors?.newPassword?.message ? 'asterisk' : null}
              intent="danger"
            />
          </InputWrap>
          <InputWrap>
            <InputGroup
              {...register('confirmPassword')}
              type="password"
              label="Confirm New Password"
              placeholder="Confirm New Password"
              errorMessage={errors?.confirmPassword?.message}
              icon={errors?.confirmPassword?.message ? 'asterisk' : null}
              intent="danger"
            />
          </InputWrap>
        </FormLayout>
        <ButtonGroup>
          <Button intent="primary" type="submit">
            Save Changes
          </Button>
        </ButtonGroup>
      </form>
    )
  } else {
    return (
      <>
        <BodyText hasGreyText>Please use the Reapit Connect My Account app to manage your account</BodyText>
        <ButtonGroup className={elMb11}>
          <Button intent="primary" type="submit" onClick={openNewPage(`${getTokenIssuer(token)}my-account`)}>
            Manage Account
          </Button>
        </ButtonGroup>
      </>
    )
  }
}
