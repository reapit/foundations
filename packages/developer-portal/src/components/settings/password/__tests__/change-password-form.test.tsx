import React from 'react'
import { ChangePasswordForm, handleChangePassword } from '../change-password-form'
import { render } from '../../../../tests/react-testing'
import { changePasswordService } from '../../../../services/cognito-identity'
import { UseSnack } from '@reapit/elements'

jest.mock('../../../../services/cognito-identity', () => ({
  changePasswordService: jest.fn(),
}))

const mockChangePasswordService = changePasswordService as jest.Mock

describe('ChangePasswordForm', () => {
  it('should match snapshot', () => {
    expect(render(<ChangePasswordForm />)).toMatchSnapshot()
  })
})

describe('handleChangePassword', () => {
  it('should successfully change the password', async () => {
    mockChangePasswordService.mockReturnValue(true)
    const email = 'MOCK_EMAIL'
    const snack = {
      success: jest.fn(),
      error: jest.fn(),
    } as unknown as UseSnack
    const formValues = {
      password: 'MOCK_PASSWORD',
      newPassword: 'MOCK_NEW_PASSWORD',
      confirmPassword: 'MOCK_NEW_PASSWORD',
    }

    const curried = handleChangePassword(email, snack)

    await curried(formValues)

    expect(mockChangePasswordService).toHaveBeenCalledWith({
      password: formValues.password,
      newPassword: formValues.newPassword,
      userName: email,
    })

    expect(snack.error).not.toHaveBeenCalled()
    expect(snack.success).toHaveBeenCalledTimes(1)
  })

  it('should throw an error if the change password service fails', async () => {
    mockChangePasswordService.mockReturnValue(false)
    const email = 'MOCK_EMAIL'
    const snack = {
      success: jest.fn(),
      error: jest.fn(),
    } as unknown as UseSnack
    const formValues = {
      password: 'MOCK_PASSWORD',
      newPassword: 'MOCK_NEW_PASSWORD',
      confirmPassword: 'MOCK_NEW_PASSWORD',
    }

    const curried = handleChangePassword(email, snack)

    await curried(formValues)

    expect(mockChangePasswordService).toHaveBeenCalledWith({
      password: formValues.password,
      newPassword: formValues.newPassword,
      userName: email,
    })

    expect(snack.error).toHaveBeenCalledTimes(1)
    expect(snack.success).not.toHaveBeenCalled()
  })
})
