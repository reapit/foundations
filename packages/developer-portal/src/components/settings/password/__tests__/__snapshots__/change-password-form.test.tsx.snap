// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordForm should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ecbqyq6-el-snack-holder"
      />
      <form>
        <div
          class="e1fwulnz-el-form-layout-has-margin ee6e7by-el-form-layout"
        >
          <div
            class="e1ksxx4v-el-input-wrap"
          >
            <div
              class="e1979rmu-el-input-group"
            >
              <input
                class="e1urjlvd-el-input"
                id="test-static-id"
                name="password"
                type="password"
              />
              <label
                class="e1nbkkrq-el-label"
                for="test-static-id"
              >
                Current Password
              </label>
            </div>
          </div>
          <div
            class="e1ksxx4v-el-input-wrap"
          >
            <div
              class="e1979rmu-el-input-group"
            >
              <input
                class="e1urjlvd-el-input"
                id="test-static-id"
                name="newPassword"
                type="password"
              />
              <label
                class="e1nbkkrq-el-label"
                for="test-static-id"
              >
                New Password
              </label>
            </div>
          </div>
          <div
            class="e1ksxx4v-el-input-wrap"
          >
            <div
              class="e1979rmu-el-input-group"
            >
              <input
                class="e1urjlvd-el-input"
                id="test-static-id"
                name="confirmPassword"
                type="password"
              />
              <label
                class="e1nbkkrq-el-label"
                for="test-static-id"
              >
                Confirm New Password
              </label>
            </div>
          </div>
        </div>
        <div
          class="eay05y4-el-button-group"
        >
          <div
            class="ebqwrow-el-button-group-inner"
          >
            <button
              class="ebee6r1-el-intent-primary eqfkc7t-el-button"
              type="submit"
            >
              Save Changes
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <div
      class="ecbqyq6-el-snack-holder"
    />
    <form>
      <div
        class="e1fwulnz-el-form-layout-has-margin ee6e7by-el-form-layout"
      >
        <div
          class="e1ksxx4v-el-input-wrap"
        >
          <div
            class="e1979rmu-el-input-group"
          >
            <input
              class="e1urjlvd-el-input"
              id="test-static-id"
              name="password"
              type="password"
            />
            <label
              class="e1nbkkrq-el-label"
              for="test-static-id"
            >
              Current Password
            </label>
          </div>
        </div>
        <div
          class="e1ksxx4v-el-input-wrap"
        >
          <div
            class="e1979rmu-el-input-group"
          >
            <input
              class="e1urjlvd-el-input"
              id="test-static-id"
              name="newPassword"
              type="password"
            />
            <label
              class="e1nbkkrq-el-label"
              for="test-static-id"
            >
              New Password
            </label>
          </div>
        </div>
        <div
          class="e1ksxx4v-el-input-wrap"
        >
          <div
            class="e1979rmu-el-input-group"
          >
            <input
              class="e1urjlvd-el-input"
              id="test-static-id"
              name="confirmPassword"
              type="password"
            />
            <label
              class="e1nbkkrq-el-label"
              for="test-static-id"
            >
              Confirm New Password
            </label>
          </div>
        </div>
      </div>
      <div
        class="eay05y4-el-button-group"
      >
        <div
          class="ebqwrow-el-button-group-inner"
        >
          <button
            class="ebee6r1-el-intent-primary eqfkc7t-el-button"
            type="submit"
          >
            Save Changes
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
